reverse_proxy.go

```diff
@@ -487,7 +506,14 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 	}
 
 	addrs := requestIPHops(req)
-	outreq.Header.Set("X-Forwarded-For", addrs)
+	forwadedForHdr := "X-Forwarded-For"
+	if p.CheckHeaderAllowed(forwadedForHdr, p.TykAPISpec, req) {
+		log.Debug("Setting HDR: ", forwadedForHdr)
```

Comment:
Do not forget to remove log debug calls, and I think you can pass header name directly instead of using `forwadedForHdr` variable. Also, I think you can remove `outreq.Header.Del` at all, because our header transformation middleware will delete it, so you do the double work: goal of this functionality is not to override/add new heaver value.